# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [22]:
import pandas as pd
import numpy as np
from scipy import stats

In [23]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [24]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [25]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

### 1. What test is appropriate for this problem? Does CLT apply?
Since the population variance is unknown, we can use t-test for this problem.
Yes CLT is valid here. We have a large sample size.

### 2. What are the null and alternate hypotheses?
Null hypothesis is that race does not have a significant impact on the rate of callbacks for resumes.
Alternative hypothesis is that race has a significant impact on the rate of callbacks for resumes.

### 3. Compute margin of error, confidence interval, and p-value.


In [26]:
df = data[["race", "call"]]
df.head()

race  call
0    w   0.0
1    w   0.0
2    b   0.0
3    b   0.0
4    w   0.0

In [47]:
df_b = df.loc[df["race"]=="b"]
df_w = df.loc[df["race"]=="w"]

# Proportion of call for blacks:
prop_call_b = np.sum(df_b.call == 1)/df_b.call.count()
print("Proportion of call for blacks: ", prop_call_b)

# Proportion of call for whites:
prop_call_w = np.sum(df_w.call == 1)/df_w.call.count()
print("Proportion of call for whites: ", prop_call_w)

# Observed difference  
diff_prop_obs = prop_call_w - prop_call_b
print("Difference in proportion of call backs: ", diff_prop_obs)

# Standard Error
s_error = np.sqrt(df_b["call"].var()/df_b["call"].count()+ df_w["call"].var()/df_w["call"].count()) 
print("Standard Error: ", s_error)

# Margin of erroe for 95% confidence
margin_error = s_error*1.96
print("Margin of Error: ", margin_error)

# T-score
t_stat, p = stats.ttest_ind(df_b['call'],df_w['call'],equal_var=False)
print('t-statistic: ', np.abs(t_stat))
print('p-value: ', p)

# Confidence Intervals
low_conf_int = diff_prop_obs - 1.96*s_error
high_conf_int = diff_prop_obs + 1.96*s_error
print('Confidence Intervals: ', low_conf_int, high_conf_int)


Proportion of call for blacks:  0.064476386037
Proportion of call for whites:  0.0965092402464
Difference in proportion of call backs:  0.0320328542094
Standard Error:  0.00778490682688
Margin of Error:  0.0152584173807
t-statistic:  4.11470529086
p-value:  3.94294151365e-05
Confidence Intervals:  0.0167744368288 0.0472912715901


#### Since the p-value is smaller than 0.05, we reject the null hypothesis. Therefore there is a statistical significance between the number of call-backs for resumes with black and white sounding names.

### 4. Write a story describing the statistical significance in the context or the original problem.
#### Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.
#### In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.
#### We wanted to determine if the race has any effect on the number of callbacks from the employers. Looking at the provided data, we see that proportion of callbacks for black people is lower than white people (0.064476386037 vs 0.0965092402464 respectively). But is this due to random or is statistically significant. We set a null hypothesis that there is no difference between number of calls backs from black and white resumes. Then we calculate t-score and p-value. High t-statistic and low p-value ( 3.94294151365e-05) shows that the two samples are statistically significantly different and therefore we reject the null hypothesis. 

### 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?
Our analysis has shown that race/name is an important factor in callback. This does not mean that it is the most important factor. In order to determine that we need to do hypothesis testings on other variable and determine if there is a statistically significant difference between callbacks for those specific variables.